python/lsst/fgcmcal/fgcmBuildFromIsolatedStars.py

```diff
@@ -0,0 +1,858 @@
+        }
+
+        if len(isolated_star_cat_handle_dict) != len(isolated_star_source_handle_dict):
+            raise RuntimeError("isolated_star_cats and isolate_star_sources must have same length.")
+
+        for tract in isolated_star_cat_handle_dict:
+            if tract not in isolated_star_source_handle_dict:
+                raise RuntimeError(f"tract {tract} in isolated_star_cats but not isolated_star_sources")
+
+        if self.config.doReferenceMatches:
+            lut_handle = input_ref_dict["fgcm_lookup_table"]
+
+            # Prepare the reference catalog loader
+            ref_config = LoadReferenceObjectsConfig()
+            ref_config.filterMap = self.config.fgcmLoadReferenceCatalog.filterMap
+            ref_obj_loader = ReferenceObjectLoader(dataIds=[ref.datasetRef.dataId
+                                                            for ref in inputRefs.ref_cat],
+                                                   refCats=butlerQC.get(inputRefs.ref_cat),
+                                                   name=self.config.connections.ref_cat,
+                                                   log=self.log,
+                                                   config=ref_config)
+            self.makeSubtask('fgcmLoadReferenceCatalog',
+                             refObjLoader=ref_obj_loader,
+                             refCatName=self.config.connections.ref_cat)
+        else:
+            lut_handle = None
+
+        # The visit summary handles for use with fgcmMakeVisitCatalog must be keyed with
+        # visit, and values are a list with the first value being the visit_summary_handle,
+        # the second is the source catalog (which is not used, but a place-holder is
+        # left for compatibility reasons.)
+        visit_summary_handle_dict = {handle.dataId['visit']: [handle, None] for
+                                     handle in input_ref_dict['visit_summaries']}
+
+        camera = input_ref_dict["camera"]
+
+        struct = self.run(
+            camera=camera,
+            visit_summary_handle_dict=visit_summary_handle_dict,
+            isolated_star_cat_handle_dict=isolated_star_cat_handle_dict,
+            isolated_star_source_handle_dict=isolated_star_source_handle_dict,
+            source_schema=self.source_schema,
+            lut_handle=lut_handle,
+        )
+
+        butlerQC.put(struct.fgcm_visit_catalog, outputRefs.fgcm_visit_catalog)
+        butlerQC.put(struct.fgcm_star_observations, outputRefs.fgcm_star_observations)
+        butlerQC.put(struct.fgcm_star_ids, outputRefs.fgcm_star_ids)
+        if self.config.doReferenceMatches:
+            butlerQC.put(struct.fgcm_reference_stars, outputRefs.fgcm_reference_stars)
+
+    def run(self, camera=None, visit_summary_handle_dict=None, isolated_star_cat_handle_dict=None,
+            isolated_star_source_handle_dict=None, source_schema=None, lut_handle=None):
```

Comment:
Cool, I didn't know about this.

Reply:
They aren't required keywords at all because the `=None` there. If they aren't allowed to have default values then can we remove the defaults and put the `*` at the start as well? (to force them to be keyword args).

Reply:
(which I see is what you did just as I pushed the button here).